packages/logger/src/Logger.ts

```diff
@@ -446,11 +463,12 @@ class Logger extends Utility implements LoggerInterface {
 
   public static injectLambdaContextAfterOrOnError(
     logger: Logger,
-    initialPersistentAttributes: LogAttributes,
+    persistentAttributes: LogAttributes,
     options?: InjectLambdaContextOptions
   ): void {
     if (options && options.clearState === true) {
-      logger.setPersistentLogAttributes(initialPersistentAttributes);
+      logger.clearState();
+      logger.addPersistentLogAttributes(persistentAttributes);
```

Comment:
We can remove the `logger.addPersistentLogAttributes(persistentAttributes);` line/call, and mark this method as `@deprecated`.